# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../collectd/out-of-bounds.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- collectd-4.4.0/src/teamspeak2.c	2008-05-06 16:14:06.000000000 +0200
+++ collectd-4.4.0-fixed/src/teamspeak2.c	2008-05-13 10:03:40.000000000 +0200
@@ -389,7 +389,7 @@
 		ERROR ("teamspeak2 plugin: tss2_receive_line failed.");
 		return (-1);
 	}
-	response[sizeof (response)] = 0;
+	response[sizeof (response) - 1] = 0;
 
 	/* Check answer */
 	if ((strncasecmp ("OK", response, 2) == 0)
@@ -437,7 +437,7 @@
 			ERROR ("teamspeak2 plugin: tss2_receive_line failed.");
 			return (-1);
 		}
-		buffer[sizeof (buffer)] = 0;
+		buffer[sizeof (buffer) - 1] = 0;
 		
 		if (strncmp ("average_packet_loss=", buffer,
 					strlen ("average_packet_loss=")) == 0)
